src/core/filesystem.cc

```diff
@@ -1342,14 +1342,9 @@ S3FileSystem::IntializeAndCheckClient(const std::string& s3_path)
         /*useVirtualAdressing*/ false);
   }
 
-  // Verify that bucket exists and 'client_' is initialized correctly.
-  bool is_dir;
-  RETURN_IF_ERROR(IsDirectory(clean_path, &is_dir));
-  if (!is_dir) {
-    return Status(
-        Status::Code::INVALID_ARG, "No bucket/folder found at " + clean_path +
-                                       ". Check account credentials.");
-  }
+  // Verify that the path exists and 'client_' is initialized correctly.
+  bool exists;
+  RETURN_IF_ERROR(FileExists(clean_path, &exists));
```

Comment:
> if (!head_object_outcome.IsSuccess()) {
>     return Status(
>         Status::Code::INTERNAL,
>         "Could not get MetaData for object at " + path + " due to exception: " +
>             head_object_outcome.GetError().GetExceptionName() +
>             ", error message: " + head_object_outcome.GetError().GetMessage());
>   } else {
>     *exists = true;
>   }
I believe we don't need exists output parameter here. Either the `exists` is set true or the FileExists returns an erroneous status.

Reply:
We need to return that the file exists. The file can be a directory,. The flow is such:
1.  Check if Path is a valid Path (Can be parsed correctly)
2. Check if path is a directory. If directory then exists is set to True and return success, else it is set to False. 
3. Reach here if not a directory. Check if path is a file. If so, exists is set to True.